### PR TITLE
✨ Add alternative prices url

### LINKED ISSUE #274: Alternative prices.tf URL
As a user I'd like to use an alternative prices.tf URL.

Feature: user can set alternative autoprice source other than prices.tf

### PR DESCRIPTION
**Pull request by joekiller**
Set CUSTOM_PRICER_URL="" to a url other than https://api.prices.tf to use a different price source.
PRICESTF_API_TOKEN renamed to CUSTOM_PRICER_API_TOKEN

Closes #274

### CODE REVIEW

File: template.env [hunk 1]
```diff
@@ -14,6 +14,7 @@ GROUPS=["103582791469033930"]
 ALERTS=["trade"]
 
 PRICESTF_API_TOKEN=""
+PRICESTF_URL=""
```
Comment (idinium96) [inline template.env:17]: maybe best to just change these two to other name, such as, `CUSTOM_PRICER_URL` and `CUSTOM_PRICER_API_TOKEN`

File: src/classes/BotManager.ts [hunk 2]
```diff
@@ -30,11 +32,12 @@ export default class BotManager {
 
     private exiting = false;
 
-    constructor(pricestfApiToken?: string) {
-        this.schemaManager = new SchemaManager({});
+    constructor(pricestfUrl?: string, pricestfApiToken?: string) {
+        this.priceSource = new PricesTf(pricestfUrl, pricestfApiToken);
+        this.schemaManager = new SchemaManager({ apiKey: pricestfApiToken });
```
[the remark this answers is not in the document]
Comment (joekiller) [reply src/classes/BotManager.ts:37]: Okay I understand now

File: template.env (hunk 1 above)
Comment (joekiller) [reply template.env:17]: Yes. Thanks for the feedback

File: src/classes/Pricer.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    source: string;
+    time: number;
+    buy: KeysMetal | null;
+    sell: KeysMetal | null;
+}
+
+export interface KeysMetal {
+    keys: number;
+    metal: number;
+}
```
Comment (idinium96) [inline src/classes/Pricer.ts:79]: This can be replaced by `Currencies` by importing it from `tf2-currencies-2` library.

File: tsconfig.json [hunk 4]
```diff
@@ -1,5 +1,6 @@
 {
     "compilerOptions": {
+        "baseUrl": ".",
```
Comment (idinium96) [inline tsconfig.json:3]: Is this need to be edited by the user? If so, then tsconfig.json need to be included in `.gitignore` file I guess.

Comment (idinium96) [reply tsconfig.json:3]: or people that want to use a custom pricer will need to fork the repo in order to edit this file?

Comment (idinium96) [reply tsconfig.json:3]: well but then, other people will have trouble compiling the code because tsconfig.json file will never be found.

Comment (joekiller) [reply tsconfig.json:3]: no this just says, the project starts from this tsconfig location. it shouldn't need to be editted.

**Comment by joekiller** [comment]
I'm good with this feature for now.

File: tsconfig.json (hunk 4 above)
Comment (idinium96) [reply tsconfig.json:3]: I see. noted

**Comment by joekiller** [comment]
There is a bug in this still which I need to fix. Occurs after most transactions.

**Comment by idinium96** [comment]
Noted sir. 👌

**Comment by joekiller** [comment]
https://github.com/TF2Autobot/tf2autobot/commit/b967960f6bc0cd46dc38d40fe66669906b73250f fixed the bug I was tracking so I think this is good now. I've been running on this branch with 3.0.1 master merged and 3.0 prior and beta merged prior etc. just did a rebase.

**Comment by joekiller** [comment]
nope still there. I was just hoping b/c it looked like it. I'll chase it down...

**Comment by idinium96** [comment]
image

**Comment by joekiller** [comment]
Okay dokie, I'm testing this on my bots. No errors so far. I pulled out anything in the interface that wasn't actually used so that functionality isn't misleading.

**Comment by idinium96** [comment]
Alrighty, sir!

**Comment by joekiller** [comment]
This seems ready. 🚀

**Comment by idinium96** [comment]
If I merge this, then it should be v3.2.0.. so fast ⚡

**Review by idinium96 — APPROVED** [review]
Looks good although never tested it. Maybe good to merge with `development` and test this right away.

**Pull request merged**